Ftd207 ultra ram

andrewvanderbyl opened the pull request:
Added URAM option to delay_wideband_prog block.

andrewvanderbyl committed:
Merge branch 'm2021a' of github.com:ska-sa/mlib_devel into m2021a

andrewvanderbyl committed:
Added UltraRAM and Block RAM option to delay_bram_prog.

andrewvanderbyl committed:
Fixed memory type selection.

andrewvanderbyl committed:
Added 'Ultra RAM' option to top level parameters of 'delay_wideband_prog' block.

Known Issue: Memory has to be 'dual port' for BRAM and Ultra RAM selection. Chnaging to 'single port' generates an error.
Also - no support for 'Distributed memory' added - this would require modification of other scripts. If 'Distributed memory'
is required, it would be necessary to comment out the init script and manually set the memory type in the block.

amartens reviewed (approved):
The Ultra RAM option does not find its way down when a single port RAM option is chosen (yet the single port RAM block does have it as an option). It does work as expected for the Dual Port RAM option though, so I think this is useful as-is.

amartens committed:
Merge branch 'm2021a' into ftd207_ultraRAM

Merging with -X ours as casper_library_delays won't merge right now as
m2021a doesn't have delay_bram changes. I have saved these changes
separately and will re-add it manually after this.

amartens committed:
re-add of delay_bram with UltraRAM option

slx files are binary, so previous merge did not work as branch was
made pre delay_bram update. So re-did the UltraRAM option to delay_bram.

delay_wideband_prog and fft_wideband_real still seem to initialise correctly.

amartens commented:
I had to re-add the UltraRAM option added to the delay_bram block as casper_library_delays wouldn't merge.
